packages/shared/src/server/automations/webhooks.ts

```diff
@@ -0,0 +1,265 @@
+import { Job, Processor } from "bullmq";
+import { backOff } from "exponential-backoff";
+import {
+  ActionExecutionStatus,
+  JobConfigState,
+} from "../../../prisma/generated/types";
+import {
+  PromptWebhookOutboundSchema,
+  WebhookDefaultHeaders,
+} from "../../domain";
+import { prisma } from "../../db";
+import { TQueueJobTypes, QueueName, WebhookInput } from "../queues";
+import {
+  getActionByIdWithSecrets,
+  getAutomationById,
+  getConsecutiveAutomationFailures,
+} from "../repositories";
+import { logger } from "..";
+import { createSignatureHeader } from "../../encryption/signature";
+import { decrypt } from "../../encryption";
+import { InternalServerError, LangfuseNotFoundError } from "../../errors";
+
+export const webhookProcessor: Processor = async (
+  job: Job<TQueueJobTypes[QueueName.WebhookQueue]>,
+) => {
+  try {
+    return await executeWebhook(job.data.payload);
+  } catch (error) {
+    logger.error("Error executing WebhookJob", error);
+    throw error;
+  }
+};
+
+// TODO: Webhook outgoing API versioning
+export const executeWebhook = async (input: WebhookInput) => {
+  const executionStart = new Date();
+
+  const { projectId, automationId, executionId } = input;
+  let httpStatus: number | undefined;
+  let responseBody: string | undefined;
+
+  try {
+    logger.debug(`Executing webhook for automation ${automationId}`);
+
+    const automation = await getAutomationById({
+      projectId,
+      automationId,
+    });
+
+    if (!automation) {
+      throw new LangfuseNotFoundError(`Automation ${automationId} not found`);
+    }
+
+    const actionConfig = await getActionByIdWithSecrets({
+      projectId,
+      actionId: automation.action.id,
+    });
+
+    if (!actionConfig) {
+      throw new Error("Action config not found");
+    }
+
+    if (actionConfig.config.type !== "WEBHOOK") {
+      throw new InternalServerError("Action config is not a webhook");
+    }
+
+    // TypeScript now knows actionConfig.config is WebhookActionConfig
+    const webhookConfig = actionConfig.config;
+
+    // Validate that the webhook URL uses HTTPS protocol for security
+    try {
+      const webhookUrl = new URL(webhookConfig.url);
+      if (webhookUrl.protocol !== "https:") {
+        throw new Error(
+          `Webhook URL must use HTTPS protocol for security. Received: ${webhookUrl.protocol}`,
+        );
+      }
```

Comment:
IMO not necessary to enforce that here. I would show a warning on the client/frontend, but if someone wants to receive the call via HTTP they should be able to. As long as we don't transfer anything sensitive this makes it easy.